app/js/lib/foxbox/recipes.js

```diff
@@ -24,232 +49,222 @@ export default class Recipe {
+            }]);
+            break;
         }
-      ))
-      // User friendly getters.
-      .then(services => services.map(service => {
-          switch (service.adapter) {
-            case 'espeak_adapter@link.mozilla.org':
-              service.setters = [
-                {
-                  label: '"Good morning!"',
-                  value: 'espeak@link.mozilla.org,Sentence,String,' +
-                  '"Good morning!"'
-                },
-                {
-                  label: '"Good afternoon!"',
-                  value: 'espeak@link.mozilla.org,Sentence,String,' +
-                  '"Good afternoon!"'
-                },
-                {
-                  label: '"Good evening!"',
-                  value: 'espeak@link.mozilla.org,Sentence,String,' +
-                  '"Good evening!"'
-                }
-              ];
-              break;
-
-            case 'ip-camera@link.mozilla.org':
-              service.setters = [
-                {
-                  label: 'takes a picture',
-                  value: 'setter:snapshot.ae67e622-7a66-465e-bab0-' +
-                  'b0c5540c5748@link.mozilla.org,TakeSnapshot,null,null'
-                }
-              ];
-              break;
-
-            default:
-              //service.setters = service.setters;
-              break;
-          }
 
-          return service;
+        return {
+          id: getter.id,
+          kind: getter.kind,
+          name: name || getter.adapter,
+          options
+        };
+      });
+    });
+  }
+
+  getSetters() {
+    // Currently we support only Camera and TTS as actuator.
+    const supportedKinds = [
+      'TakeSnapshot',
+      {
+        vendor: 'team@link.mozilla.org',
+        adapter: 'eSpeak adapter',
+        kind: 'Sentence',
+        type: 'String'
+      }
+    ];
+
+    const settersURL = `${this[p.net].origin}/api/` +
+      `v${this[p.settings].apiVersion}/channels/setters`;
+
+    return this[p.net].fetchJSON(
+      settersURL, 'POST', supportedKinds.map(kind => ({ kind }))
+    )
+    .then(setters => {
+      return setters.map(setter => {
+        let name, options = [];
+
+        // Check for the complex extension kind
+        if (typeof setter.kind === 'object') {
+
+          // WORKAROUND: waiting for the Taxonomy fix so that we have "type" for
+          // all places.
+          setter.kind.type = setter.kind.typ;
+          delete setter.kind.typ;
+
+          if (setter.kind.kind === 'Sentence') {
+            name = 'say';
+            options.push(...[{
+              label: '"Good morning!"',
+              value: { String: '"Good morning!"' }
+            }, {
+              label: '"Good afternoon!"',
+              value: { String: '"Good afternoon!"' }
+            }, {
+              label: '"Good evening!"',
+              value: { String: '"Good evening!"' }
+            }]);
+          }
+        } else if (setter.kind === 'TakeSnapshot') {
+          name = 'Camera';
```

Comment:
Here is one issue I'm going to fix in the next PR - I'd rely on tags that are assosiated with this setter to get name (```Kitchen```, ```Front Door``` etc.)